src/compute-client/src/controller/instance.rs

```diff
@@ -797,31 +811,36 @@ where
                 compute_net.push((key, update));
             } else {
                 // Storage presumably, but verify.
-                storage_todo
-                    .entry(key)
-                    .or_insert_with(ChangeBatch::new)
-                    .extend(update.drain())
+                if self.storage_controller.collection(key).is_ok() {
```

Comment:
Did you make that change for a specific reason? It does seem to be more inline with the comment above.

Reply:
Yes, we were getting a panic there by forwarding `key`s to the storage controller that should not be known to it.

Reply:
Oh, any idea how this happened? Are we dropping collections but continue to receive write frontier updates for them?

Reply:
Documenting our discussion on Slack: if we incorrectly drop compute collection metadata too early, we would delegate to collection search to storage and storage would panic for not knowing the collection. We found the panic to be too aggressive a behavior, but wanted to make sure this got logged as an error (done now).